app/controllers/api/projects_controller.rb

```diff
@@ -22,6 +22,48 @@ def searchcode
     render json: Project.visible.where('updated_at > ?', 1.day.ago).order(:repository_url).pluck(:repository_url).compact.reject(&:blank?)
   end
 
+  # returns any updated projects in a time window, the caller can
+  # then decide which ones it needs to refetch.
+  def updated
+    # there's no ActionController::Forbidden
+    raise ActionController::BadRequest unless internal_api_key?
+
+    start = date_parameter(:start_time)
```

Comment:
I think I'm with Keith and this feels pretty convoluted and non-railsy to me

```suggestion
    start = DateTime.iso8601(params.require[:start_time])
    end = if params[:end_time].nil?
                 DateTime.current
              else
                DateTime.iso8601(params[:end_time])
```

I don't think we need to worry about strong validation of the params being "good" as it's not something we generally worry about